lib/puppet_blacksmith/forge.rb

```diff
@@ -24,37 +28,61 @@ def initialize(username = nil, password = nil, url = nil)
         puts "Ignoring url entry in .puppetforge.yml: must point to the api server at #{PUPPETLABS_FORGE}, not the Forge webpage"
         self.url = PUPPETLABS_FORGE
       end
+      self.forge_type = forge_type unless forge_type.nil?
+      unless [FORGE_TYPE_PUPPET, FORGE_TYPE_ARTIFACTORY].include?(self.forge_type)
```

Comment:
Would it be better to have a single array that stores a list of all supported `forge_types` that the `.include?` can check for rather than defining a constant for each supported `forge_type`?

Reply:
Would it be better to implement that now within @bodgit's PR or go ahead and merge this and add the Error and forge_type array in a separate PR?

Reply:
I'd be fine with merging this, but I've totally forgotten to test this PR.

Reply:
@dhollinger can you PR your change back to this PR for @bodgit to merge?

Reply:
@cdenneen I'll take a look at it tomorrow

Reply:
I'm not sure this matters, the array is only used once for validation but the constants are referred to in multiple places to tweak the URL structure, authentication headers, etc. between the different forge types.

Agree with failing instead of warning for an invalid forge type if that's what is wanted.

Reply:
I think @dhollinger suggested `SUPPORTED_FORGE_TYPES = [FORGE_TYPE_PUPPET, FORGE_TYPE_ARTIFACTORY]`. At least, that's how I interpreted it and makes sense to me.

If you could implement the hard failure I think this should be ready to merge.

Reply:
I will make both changes if it can push this over the line.

Reply:
Both comments addressed I think, just letting Travis do its thing.